File: java/short-url/src/main/java/com/example/url/entity/UrlEntity.java
package com.example.url.entity;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class UrlEntity {

    /**
     * 短链接地址
     */
    private String shortUrl;

    /**
     * 长链接地址
     */
    private UniqueUrl uniqueUrl;
}

File: backend/short-url/src/url/url.entity.ts
import {
  Column,
  CreateDateColumn,
  Entity,
  Index,
  Unique,
  PrimaryGeneratedColumn,
  UpdateDateColumn,
} from 'typeorm';

@Entity('url')
@Unique('u_shortUrl', ['shortUrl'])
@Index('i_originUrl', ['originUrl'])
export class UrlEntity {
  @PrimaryGeneratedColumn("increment")
  id: number;

  @Column({
    type: 'varchar',
    length: 200,
    nullable: false,
    comment: '原Url',
  })
  originUrl: string;

  @Column({
    type: 'varchar',
    length: 100,
    nullable: true,
    comment: '私钥',
  })
  shortUrl: string;

  @CreateDateColumn()
  createAt: Date;

  @UpdateDateColumn()
  updateAt: Date;
}
